h5g_group.go

```diff
@@ -145,6 +173,17 @@ func (g *CommonFG) ObjectNameByIndex(idx uint) (string, error) {
 	return C.GoString(&name[0]), nil
 }
 
+// ObjectTypeByIndex returns the type of the object at idx.
+func (g *CommonFG) ObjectTypeByIndex(idx uint) (GType, error) {
+	cidx := C.hsize_t(idx)
+	gtyp := GType(C.H5Gget_objtype_by_idx(g.id, cidx))
```

Comment:
Apparently this is deprecated in 1.8 in favour of H5Oget_info (see https://support.hdfgroup.org/HDF5/doc/RM/RM_H5G.html#Group-GetObjTypeByIdx).

Reply:
indeed.
I was aiming at the most common denominator.

I am not sure we have an explicit policy here:
- should we follow deprecated notices and thus tighly follow the latest 1.X releases of C-HDF5?
- or rather be the most conservative?
- or give up and provide a way to select which API to wrap (adding more work on our side to maintain multiple APIs and dropping the `go-get`-able aspect of that library. granted that with `Cgo` in the mix, `go-get`-ablity is always a somewhat fluid notion...)

alternatively, we could also vendor the C-HDF5 library we wrap.

Reply:
vendoring the hdf5 would be better I think

Reply:
@kortschak do you have recommandations about a policy?
(as noted in the thread on `gonum-dev`, vendoring is a rather big chunk of work - for which I wouldn't have the cycles to spare at the moment)

Reply:
If someone else were to do the work, we could review, but I'm not particularly inclined to do it myself. That's not a policy, but it's based on how much time there in available from maintainers at the moment.

Reply:
I'd help with the review as well.

but, concerning this PR, what should we do?

Reply:
I think we can merge this and kick the can down the road.